### ISSUE TITLE
What part of the documentation should i look at to figure out how to set source properties

### ISSUE THREAD

**Issue by AyashiNoCeres** [dnaka91/obws#13]
I want to be able update a value for a source, but none of the examples cover that...

Is there any projects that use obws that I can use to try and figure stuff out from because the examples are too basic and either I'm not looking in the right place but the documentation i find just hasn't been of much use to me...

**Comment by dnaka91** [comment]
What you're probably looking for is [set_source_settings](https://docs.rs/obws/0.7.0/obws/client/struct.Sources.html#method.set_source_settings) if by _value for a source_ is you mean the settings. 

The source settings are rather generic as there are many different ones and more can be added through plugins. Therefore, the [source_settings](https://docs.rs/obws/0.7.0/obws/requests/struct.SourceSettings.html#structfield.source_settings) field is a `Value` that you have to build yourself with `serde_json` (and optionally `serde`).

I'm currently working on a few extra structs that at least provide pre-defined data structures for all the standard sources that come with OBS, but it'll still take a while until they're complete. That should make it a bit easier to configure sources from this crate.

Current examples are just tiny examples to get started and it would be a bit overkill to provide an example for every API that exists (especially for **sources** as they would require manual setup in OBS before running the examples).

I get that the docs a not super great, they're mostly copied from the `obs-websocket` [API docs](https://github.com/Palakis/obs-websocket/blob/4.x-current/docs/generated/protocol.md). I'll try to add some more samples directly into the obws API docs in the future.

Please let me know how far you get with this information and I'm happy to provide you some examples if need be.

**Comment by AyashiNoCeres** [comment]
Hi, thanks for the reply! I eventually managed to figure out how to get it done, with a bit of time.

I think it just wasn't clear how to create those structs. Also the problem is as I'm new to rust, i'm finding that in order to understand most rust documentation that rely on autogenerated documentation I need to have a stronger understanding of rust as my eyes just glaze over when I start to see some of the source code in the libraries.

Didn't help that I was trying read and change TextGdi+ values which don't seem to be supported so I had to hack around to get that working but in the end I managed to do something like this: 

```
  let y = TextGdiPlusProperties {
    text: Some("3"),
    source: "test",
    ..TextGdiPlusProperties::default()
  };
```

Probably not the best but it gets the job done.
